lib/matching/matching.h

```diff
@@ -5,23 +5,25 @@
 
 #pragma once
 
+#include "generic/cluster_chain.h"
+#include "generic/cluster_descriptor.h"
 #include <memory>
 #include <vector>
-#include "generic/cluster_descriptor.h"
-#include "generic/cluster_chain.h"
 
 namespace MouseTrack {
 
 class Matching {
 public:
-    /// Given a list of cluster chains, we find the best matching to append the given list of descriptors.
-    /// descriptors: a list of descriptors from a new frame
-    /// chains: a list of cluster chains to which we want to appeand one cluster from descriptors
-    /// returns a vector of size descriptors.size(), which holds the index of the matching chain for each descriptor.
-    /// If no match was found, the vector value is set to -1.
-    virtual std::vector<long> operator()(const std::vector<std::shared_ptr<const ClusterDescriptor>>& descriptors, const std::vector<ClusterChain>& chains) = 0;
+  ~Matching(){};
+  /// Given a list of cluster chains, we find the best matching to append the
```

Comment:
Automatic reformatting moved line breaks which makes it hard to read, maybe just add two line breaks?